frontend/public/components/build-pipeline.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { fromNow } from './utils/datetime';
+import { K8sResourceKind } from '../module/k8s';
+
+const getBuildNumber = (resource: K8sResourceKind): number => _.get(resource, ['metadata', 'annotations', 'openshift.io/build.number']);
+const getStages = (status): any[] => (status && status.stages) || [];
+const getJenkinsStatus = (resource: K8sResourceKind) => {
+  const status = _.get(resource, ['metadata', 'annotations', 'openshift.io/jenkins-status-json']);
+  if (!status) {
+    return {};
+  }
+
+  const json = _.attempt(JSON.parse, status);
+  return _.isError(json) ? {} : json;
+};
+const getJenkinsLogURL = (resource: K8sResourceKind): string => _.get(resource, ['metadata', 'annotations', 'openshift.io/jenkins-log-url']);
+const getJenkinsBuildURL = (resource: K8sResourceKind): string => _.get(resource, ['metadata', 'annotations', 'openshift.io/jenkins-build-uri']);
+
+const BuildSummaryStatusIcon: React.SFC<BuildSummaryStatusIconProps> = ({status}) => {
+  const statusClass = _.lowerCase(status);
+  const icon = ({
+    new: 'fa-hourglass-o',
+    pending: 'fa-hourglass-half',
+    running: 'fa-refresh fa-spin',
+    complete: 'fa-check-circle',
+    failed: 'fa-times-circle'
+  })[statusClass];
+  const hide = icon ? 'hide' : '';
+  return <React.Fragment>
+    <span className={`build-pipeline__status-icon build-pipeline__status-icon--${statusClass}`}>
+      <span aria-hidden="true">
+        <span className={`fa ${icon} fa-fw`}></span>
```

Comment:
We check for empty icon above. Doesn't that mean this could result in `class="fa fa-fw"` with no icon here?

Reply:
Probably should be:

```html
  return <React.Fragment>
    {icon && <span className={`build-pipeline__status-icon build-pipeline__status-icon--${statusClass}`}>
      <span className={`fa ${icon} fa-fw`} aria-hidden="true"></span>
    </span>}

    {!icon && <span className="build-pipeline__status-icon">
      <span className="fa fa-refresh fa-spin" aria-hidden="true"></span>
    </span>}
  </React.Fragment>;
```

Reply:
Then the `<React.Fragment>` is not really needed, we will either do one or the other...

Reply:
So probably should be:

```html
  return icon
    ? <span className={`build-pipeline__status-icon build-pipeline__status-icon--${statusClass}`}>
      <span className={`fa ${icon} fa-fw`} aria-hidden="true"></span>
    </span>
    : <span className="build-pipeline__status-icon">
      <span className="fa fa-refresh fa-spin" aria-hidden="true"></span>
    </span>
```